packages/frontend/core/src/modules/right-sidebar/entities/right-sidebar.ts

```diff
@@ -0,0 +1,38 @@
+import { LiveData } from '@toeverything/infra/livedata';
+
+import type { RightSidebarView } from './right-sidebar-view';
+
+export class RightSidebar {
+  readonly isOpen = new LiveData(false);
+  readonly views = new LiveData<RightSidebarView[]>([]);
+  readonly front = this.views.map(
+    stack => stack[0] as RightSidebarView | undefined
+  );
+  readonly hasViews = this.views.map(stack => stack.length > 0);
+
+  open() {
+    this.isOpen.next(true);
+  }
+
+  toggle() {
+    this.isOpen.next(!this.isOpen.value);
+  }
+
+  close() {
+    this.isOpen.next(false);
+  }
+
+  _append(view: RightSidebarView) {
```

Comment:
Why prefixed with `_`?

Reply:
This means that this function should not be called outside this "module".
should use RightSidebarViewIsland instead.

Reply:
I have added some comment here.